src/ImageScaler.java

```diff
@@ -0,0 +1,169 @@
+            }
+            right--;
+        }
+        return image.getSubimage(left, top, right - left + 1, bottom - top + 1);
+    }
+
+    public void scalImage(String unicode, String fontName) throws InvalidValueException {
+        if (unicode.equals("")) {
+            throw new InvalidValueException("Unicode can not be empty string");
+        } else if (fontName.equals("")) {
+            throw new InvalidValueException("Font can not be empty string");
+        }
+        BufferedImage imageBeforeScal = new BufferedImage(1, 1, BufferedImage.TYPE_4BYTE_ABGR);
+        Graphics2D graphic = imageBeforeScal.createGraphics();
+        Font font = null;
+        String regex = "([A-Z|a-z]:\\\\[^*|\"<>?\\n]*)|(\\\\\\\\.*?\\\\.*)";
+        if (Pattern.matches(regex, fontName)) {
+            try {
+                font = Font.createFont(Font.TRUETYPE_FONT, new File(fontName)).deriveFont(50f);
+                GraphicsEnvironment graphicsEnviroment = GraphicsEnvironment.getLocalGraphicsEnvironment();
+                graphicsEnviroment.registerFont(Font.createFont(Font.TRUETYPE_FONT, new File(fontName)));
+            } catch (IOException | FontFormatException e) {
+                System.out.println("Wrong font format!.");
+            }
+        } else {
+            font = new Font(fontName, Font.PLAIN, 50);
+        }
+
+        graphic.setFont(font);
+        graphic.dispose();
+
+        FontMetrics fontMetrics = graphic.getFontMetrics();
+        int width = fontMetrics.stringWidth(unicode);
+        int height = fontMetrics.getHeight();
+
+        imageBeforeScal = new BufferedImage(width, height, BufferedImage.TYPE_4BYTE_ABGR);
+        graphic = imageBeforeScal.createGraphics();
+        graphic.setColor(Color.black);
+        graphic.setFont(font);
+        FontMetrics fontMetricScaleImage = graphic.getFontMetrics();
+        int x = 0;
+        int y = fontMetricScaleImage.getAscent();
+        graphic.drawString(unicode, x, y);
+        graphic.dispose();
+
+        char unicodeChar = unicode.charAt(0);
+        String hex = String.format("%04x", (int) unicodeChar);
+        File outputfile = new File(hex + ".png");
+
+        BufferedImage imageAfterScal = new BufferedImage(64, 64, BufferedImage.TYPE_4BYTE_ABGR);
+        Graphics2D graphicAfterScal = imageAfterScal.createGraphics();
+        try {
+            graphicAfterScal.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_NEAREST_NEIGHBOR);
+            imageBeforeScal = cutImage(imageBeforeScal);
+            graphicAfterScal.drawImage(imageBeforeScal, 0, 0, 64, 64, null);
+        } finally {
+            graphicAfterScal.dispose();
+        }
+        try {
+            ImageIO.write(imageAfterScal, "png", outputfile);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void scalImage(String unicode, String fontName, BufferedImage image) throws InvalidValueException {

```

Comment:
Wiem, że to kwestia opisu issue, ale tutaj ten fontName jest raczej zbędny. Wszak dostajemy już BufferedImage, który trzeba tylko przyciąć i przeskalować.